game-core/src/main/java/games/strategy/engine/lobby/PlayerEmailValidation.java

```diff
@@ -0,0 +1,26 @@
+package games.strategy.engine.lobby;
+
+import lombok.NoArgsConstructor;
+
+/** Utility class to validates user email to check if it looks valid. */
+@NoArgsConstructor
+public final class PlayerEmailValidation {
+
+  public static boolean isValid(final String emailAddress) {
+    return validate(emailAddress) == null;
+  }
+
+  /** Allow multiple fully qualified email addresses separated by spaces, or a blank string. */
+  public static String validate(final String emailAddress) {
+    final String quotedString = "\"(?:[^\"\\\\]|\\\\\\p{ASCII})*\"";
+    final String atom = "[^()<>@,;:\\\\\".\\[\\] \\x28\\p{Cntrl}]+";
+    final String word = "(?:" + atom + "|" + quotedString + ")";
+    final String subdomain = "(?:" + atom + "|\\[(?:[^\\[\\]\\\\]|\\\\\\p{ASCII})*\\])";
+    final String domain = subdomain + "(?:\\." + subdomain + ")*";
+    final String localPart = word + "(?:\\." + word + ")*";
+    final String email = localPart + "@" + domain;
+    // String regex = "(\\s*[\\w\\.-]+@\\w+\\.[\\w\\.]+\\s*)*";
+    final String regex = "(\\s*" + email + "\\s*)*";
+    return emailAddress.matches(regex) ? null : "Invalid email address";
```

Comment:
I haven't actually tried this, but is this regex actually "accurate"?
I mean this is debatable, but the best way to check if an email is actually valid is still to actually try to send one to the address.

Reply:
Funny enough, I think we had this same debate about regex the last time this code was touched. For now, it's a more significant change to remove the previous validation. Please note this is moved code from StringUtils.

I do in general agree that regex is not a good way to go, in general checking for an `@` sign and min length is almost as good and the only way to really know is to send a confirmation link to the email.

I do not rightly know if it's really that worthwhile to downgrade the regex check since it is already in place and seemingly accurate enough. Ultimately I suspect we perhaps should focus our attention elsewhere until we start to do more with email.

Reply:
I knew I discussed this piece of code some time in the past, I just couldn't recall whom I discussed this with ^^

For now I'd be fine if you just remove the comment there